### DISCUSSION TITLE
Add modern fonts with CTkFont

### DISCUSSION THREAD

**Discussion by adrienpolytechsorbonne** [TomSchimansky/CustomTkinter#936] [Q&A] [answered]
Hello everyone,

I want to know is it possible to import modern fonts from websites like Torus Notched or Circular Std Black with customtkinter it seems that it doesn't work by adding option font to CTkLabel. Thanks in advance

**Comment by adrienpolytechsorbonne** [comment]
My code for this case : 

`import tkinter
#import ttkbootstrap as ttkb
#from ttkbootstrap.constants import *
import PIL.Image, PIL.ImageTk
import customtkinter
from cv2 import (VideoCapture, namedWindow, imshow, waitKey, destroyWindow, imwrite)
import cv2


customtkinter.set_appearance_mode("Dark")
customtkinter.set_default_color_theme("dark-blue")

def button_call():
    print("Hello")

class CapgeminiYOLO:
    def __init__(self,window,size,title,logo,imglight,imgdark,video_source=0):
        self.window = window
        self.window.geometry(size)
        self.width = 1080
        self.height = 760
        self.title = title
        self.image_logo = logo
        self.imagelight = imglight
        self.imagedark = imgdark

        self.video_source = video_source
        self.vid=MyVideoCapture(self.video_source)
        self.window.iconbitmap(self.image_logo)
        self.create_frame_accueil(self.width,self.height/2,self.imagelight,self.imagedark)
        self.frame2=customtkinter.CTkLabel(master=self.window,width=self.width,height=(self.height/2)*1.25,corner_radius=20,text="")
        self.frame2.pack()
        self.frame3=customtkinter.CTkFrame(self.window,width=self.width,height=self.height,border_width=20,border_color="#000000")
        self.frame3.pack(padx=0,pady=50)
        self.button1=customtkinter.CTkButton(self.frame3,text="Testez-moi",command=button_call,text_color="#000000",bg_color="#0048ab",font=customtkinter.CTkFont(family="Circular Std Black",size=18, weight="bold"),corner_radius=10)
        self.button1.pack(padx=0,pady=0)
        self.delay=15
        self.update()
        self.window.mainloop()

    def create_frame_accueil(self,frame_width,frame_height,imglight,imgdark):
        frame1=customtkinter.CTkFrame(master=self.window,width=frame_width,height=frame_height*1.2,border_width=15,fg_color="blue",border_color="gray10")
        frame1.grid(row=0,column=0,padx=20,pady=20)
        frame1.pack()
        image = customtkinter.CTkImage(light_image=PIL.Image.open(imglight),dark_image=PIL.Image.open(imgdark),size=(618,179))
        self.create_label(frame1,"Bienvenue dans l'interface YOLO de Capgemini",image)

    def create_label(self,frame,text,img):
        label=customtkinter.CTkLabel(master=frame,width=self.width,height=self.height/4,text=text,text_color="#0048ab",corner_radius=20,bg_color="#000000",image=img,compound="bottom",font=customtkinter.CTkFont(family="Torus Notched",size=30, weight="bold"))
        label.pack()

    def update(self):
        ret,frame=self.vid.get_frame()
        if ret:
            self.photo=customtkinter.CTkImage(light_image=PIL.Image.fromarray(frame),dark_image=PIL.Image.fromarray(frame),size=(self.frame2.cget("width")/2,self.frame2.cget("height")/1.5))
            self.frame2.configure(image=self.photo)
            #self.photo = PIL.ImageTk.PhotoImage(image=PIL.Image.fromarray(frame))
            self.window.after(self.delay,self.update)

class MyVideoCapture:
	"""docstring for  MyVideoCapture"""
	def __init__(self, video_source=0):
		self.vid = VideoCapture(video_source)
		if not self.vid.isOpened():
			raise ValueError("unable open video source", video_source)

		self.width = self.vid.get(cv2.CAP_PROP_FRAME_WIDTH)
		self.height = self.vid.get(cv2.CAP_PROP_FRAME_HEIGHT)

	def get_frame(self):
		if self.vid.isOpened():
			ret, frame = self.vid.read()
			if ret:
				return (ret, cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
			else:
				return (ret, None)
		else:
			return (ret,None)		
	def __del__(self):
		if self.vid.isOpened():
			self.vid.release()

app = customtkinter.CTk()
CapgeminiYOLO(app,"1080x760","Interface YOLO Capgemini","capgeminil.ico","capg_img.png","capg_logo.png")`

**Comment by Akascape** [comment, marked as answer]
> is it possible to import modern fonts from websites like Torus Notched or Circular Std Black with customtkinter...

Yes, it's possible to import external fonts with customtkinter.
Here is an example where I used the Torus Notched font:

![Screenshot](https://user-images.githubusercontent.com/89206401/209443244-9b5372ba-2d8c-4374-b138-0811575bdd17.jpg)

(I have not installed that font in my system, but just have the **.ttf** file)
## CODE
```python
import customtkinter
  
app = customtkinter.CTk()

customtkinter.FontManager.load_font("Torus Notched SemiBold.ttf")  # Import the .ttf file by this method

# then simply use that font 
label = customtkinter.CTkLabel(app, text="Hello!", font=("Torus Notched SemiBold",30)) 
label.grid(padx=70, pady=50)

app.mainloop()
```

**Reply by adrienpolytechsorbonne** [reply]
I managed to import them after many tries thanks for your advice.

**Reply by vincent7429** [reply]
Thank you! I was wondering why pyglet and normal tkinter importing fonts was not working. I’m guessing that because now CustomTkinter has their own utility and changed the properties.

**Reply by TorMagne** [reply]
i do not get this to work. i have my font in a folder fonts/Roboto/allthefontetc.. do not understand how i import these

**Reply by Akascape** [reply]
@TorMagne Import like this:
```python
customtkinter.FontManager.load_font("fonts/Roboto/allthefont/your_font.ttf")
```

**Reply by racine890** [reply]
It's possible with Pillow, but it draws the text as an image.

` import tkinter as tk
from PIL import Image, ImageFont, ImageDraw, ImageTk

fenetre = tk.Tk()
fenetre.geometry("300x100")

chemin_police = "chemin/vers/votre/police.ttf"

ma_police = ImageFont.truetype(chemin_police, size=12)

texte = "Bonjour, monde!"
image = Image.new("RGBA", (300, 100), (255, 255, 255, 0))
draw = ImageDraw.Draw(image)
draw.text((10, 30), texte, fill="black", font=ma_police)

image_tk = ImageTk.PhotoImage(image)

label = tk.Label(fenetre, image=image_tk)
label.image = image_tk
label.pack()

fenetre.mainloop() `
